src/ir_def/construct.ml

```diff
@@ -77,8 +77,8 @@ let wildP =
 
 let varE (id, typ) =
   if T.is_mut typ
-  then { it = PrimE (MutReadPrim id, []); at = no_region; note = Note.{ def with typ = T.as_immut typ } }
-  else { it = VarE id; at = no_region; note = Note.{ def with typ } }
+  then { it = VarE (Var, id); at = no_region; note = Note.{ def with typ = T.as_immut typ } }
+  else { it = VarE (Const, id); at = no_region; note = Note.{ def with typ } }
```

Comment:
I think you could push the outer if into then VarE of the then branch, and drop the else branch.

Reply:
Indeed: 31cf9cc8a